buildvertical.targets

```diff
@@ -8,102 +8,92 @@
     <ImportedBuildVerticalTargets>true</ImportedBuildVerticalTargets>
   </PropertyGroup>
   
-  <!-- Add vertical projects to the project group for BuildAllProjects -->
-  <Target Name="AddVerticalProjects"
-          Inputs="%(Project.Identity)"
-          Outputs="fake"
-          DependsOnTargets="DetermineProjectsConfiguration;TrimUnconvertedProjects"
-          BeforeTargets="AnnotateProjectsWithConfiguration">
-    <ItemGroup>
-      <OmitProject Include="%(Project.Identity)" Condition="'%(Project.OmitReason)' != ''">
-        <OmitReason>%(Project.OmitReason)</OmitReason>
-      </OmitProject>
-      <Project Remove="@(OmitProject)" />
-    </ItemGroup>
+ <Target Name="AnnotateProjectsWithConfiguration" 
+         Returns="@(ProjectWithConfiguration)"> 
+    <PropertyGroup>
+      <_ProjectName>$(MSBuildProjectFullPath)</_ProjectName>
+    </PropertyGroup>
+    <Message Importance="Low" Text="$(MSBuildProjectFullPath), Configuration - $(_BestConfiguration)" />
+    <ItemGroup> 
+       <ProjectWithConfiguration Include="$(_ProjectName)" Exclude="@(OmitProject)"> 
+         <AdditionalProperties Condition="'$(_BestConfiguration)' != '' and '$(_BestConfiguration)' != 'None'">Configuration=$(_BestConfiguration);%(ProjectWithConfiguration.AdditionalProperties)</AdditionalProperties>
+         <UndefineProperties Condition="'$(_BestConfiguration)' == '' or '$(_BestConfiguration)' == 'None'">Configuration;%(ProjectWithConfiguration.UndefineProperties)</UndefineProperties>
+       </ProjectWithConfiguration>
+       <ProjectWithConfiguration Remove="@(OmitProject)" />
+     </ItemGroup>
   </Target>
 
   <!-- The initial shaking target for trimming down applicable projects for specified vertical -->
   <Target Name="DetermineProjectsConfiguration"
-          Inputs="%(Project.Identity)"
+          Inputs="$(MSBuildProjectFullPath)"
           Outputs="fake"
-          Returns="@(Project)"
-          DependsOnTargets="TrimUnconvertedProjects"
+          Returns="@(OmitProject)"
           BeforeTargets="AnnotateProjectsWithConfiguration">
     <PropertyGroup>
-      <_VerticalProject>%(Project.Identity)</_VerticalProject>
-      <_VerticalProjectName>%(Project.FullPath)</_VerticalProjectName>
-      <_OmitReason>%(Project.OmitReason)</_OmitReason>
+      <_VerticalProject>$(MSBuildProjectFullPath)</_VerticalProject>
     </PropertyGroup>
 
-    <!-- Clear the VerticalTargets item -->
-    <ItemGroup><VerticalTargets Remove="@(VerticalTargets)" /></ItemGroup>
+    <!-- Clear the _Configuration item -->
+    <ItemGroup><_Configuration Remove="@(_Configuration)" /></ItemGroup>
     <MSBuild Targets="FindBestConfiguration"
```

Comment:
Why do we have to use MSBuild to call this target? I would expect that we are in the project and we should just call the FindBestConfiguration task directly.